pytest_selenium/drivers/firefox.py

```diff
@@ -78,67 +88,111 @@ def firefox_options(request, firefox_path, firefox_profile):
     if firefox_path is not None:
         options.binary = FirefoxBinary(firefox_path)
 
-    args = request.node.get_marker('firefox_arguments')
-    if args is not None:
-        for arg in args.args:
-            options.add_argument(arg)
+    for arg in get_arguments_from_markers(request.node):
+        options.add_argument(arg)
 
-    prefs = request.node.get_marker('firefox_preferences')
-    if prefs is not None:
-        for name, value in prefs.args[0].items():
-            options.set_preference(name, value)
+    for name, value in get_preferences_from_markers(request.node).items():
+        options.set_preference(name, value)
 
     return options
 
 
-@pytest.fixture(scope='session')
+def get_arguments_from_markers(node):
+    # get_marker is deprecated since pytest 3.6
+    # https://docs.pytest.org/en/latest/mark.html#marker-revamp-and-iteration
+    try:
+        arguments = list()
+        for level, mark in node.iter_markers_with_node("firefox_arguments"):
+            LOGGER.debug(
+                "{0} marker <{1.name}> "
+                "contained args <{1.args}>".format(level.__class__.__name__, mark)
+            )
+            arguments += mark.args
+        LOGGER.info("Firefox arguments from markers: {}".format(arguments))
```

Comment:
As we're appending to a list I don't think it's necessary to include this debug. We included it for capabilities because we update the dictionary, which is more likely to cause confusion. We can probably just use something like:

```python
try:
    arguments = []
    [arguments.extend(m.args) for m in node.iter_markers("firefox_arguments")]
    return arguments
except AttributeError:
    arguments = node.get_marker("firefox_arguments")
    return arguments.args if arguments else []
```